go/cmd/dolt/commands/sqlserver/metrics_listener.go

```diff
@@ -88,10 +111,79 @@ func newMetricsListener(labels prometheus.Labels, versionStr string) (*metricsLi
 	prometheus.MustRegister(ml.gaugeConcurrentQueries)
 	prometheus.MustRegister(ml.histQueryDur)
 
+	go func() {
+		for ml.updateReplMetrics() {
+			time.Sleep(clusterUpdateInterval)
+		}
+	}()
+
 	ml.gaugeVersion.Set(f64Version)
 	return ml, nil
 }
 
+func (ml *metricsListener) updateReplMetrics() bool {
+	ml.mu.Lock()
+	defer ml.mu.Unlock()
+
+	if ml.done {
+		return false
+	}
+
+	perDbStatus := ml.clusterStatus.GetClusterStatus()
+	dbNames := make(map[string]struct{})
+
+	for _, status := range perDbStatus {
```

Comment:
I know this is gonna sound nitty but I think this is more natural as a set of GaugeVecs.

I think one should probably be

```go
roleVec := prometheus.NewGaugeVec(prometheus.GaugeOpts{
    Name: "dss_cluster_role",
    Help: "one if the server is currently in this role, zero otherwise",
}, []string {"database", "role"})
```

and one which is

```go
lagVec := prometheus.NewGaugeVec(prometheus.GaugeOpts{
    Name: "dss_replication_lag",
    Help: "The reported replication lag of this server when it is a primary to the given standby.",
}, []string {"database", "remote"})
```

The `lagVec` should probably be populated as something like:

```go
for _, status := range status {
    if status.Role == "primary" {
        lagVec.WithLabelValues(status.Database, status.Remote).Set(status.ReplicationLag.Milliseconds())
    } else {
        lagVec.WithLabelValues(status.Database, status.Remote).Set(-1.0)
   }
}
```

or something like that. The roleVec should probably be populated:

```
for _, role := range []string{"primary", "standby", "detected_broken_config"} {
    for _, status := range status {
        if status.Role == role {
            roleVec.WithLabelValues(status.Database, role).Set(1.0)
        } else {
            roleVec.WithLabelValues(status.Database, role).Set(0.0)
        }
    }
}
```

(or `[]string{string(cluster.RolePrimary), string(cluster.RoleStandby), string(cluster....)}`)

Or some such.

The set of databases currently reported can be kept between runs if you want. At the end, any database in the prior set which was not seen can be removed with:

```go
for _, db := range ml.clusterSeenDBs {
    if _, ok := newSeenDBs[db]; !ok {
        lagVec.DeletePartialMatch(prometheus.Labels{"database": db})
        roleVec.DeletePartialMatch(prometheus.Labels{"database": db})
    }
}
ml.clusterSeenDBs = newSeenDBs
```

or some such.